frappe/email/doctype/email_alert/email_alert.py

```diff
@@ -21,6 +23,18 @@ def validate(self):
 			frappe.throw(_("Please specify which value field must be checked"))
 
 		self.validate_forbidden_types()
+		self.validate_condition()
+		
+
+	def validate_condition(self):
+		temp_doc = frappe.new_doc(self.document_type)	
+		context = {"doc": temp_doc, "nowdate": nowdate}
+		if self.condition:
+			try:
+				eval(self.condition, context)
+			except:
+				frappe.throw("Please check your Condition")
```

Comment:
Should be translated. Also I think, 'invalid' is a better word than 'please check'. What do you think?

`frappe.throw(_('The Condition is invalid'))`

Also, commonify the **context** and use it instead of defining `context = ...` where it is used. This is more maintainable. 

```
def get_context(self, doc):
    return {'doc': doc, 'nowdate': nowdate}
```